crates/sui-types/src/temporary_store.rs

```diff
@@ -1141,6 +1143,15 @@ impl<S: ObjectStore> TemporaryStore<S> {
 // Charge gas current - end
 //==============================================================================
 
+impl<S: ObjectStore> TemporaryStore<S> {
+    pub fn advance_epoch_safe_mode(&mut self, params: &AdvanceEpochParams) {
```

Comment:
This seems like an oddly specific function to go on TemporaryStore, can it not just be a free function that accepts the store as a parameter?

Reply:
It needs mutable access to the private store of TemporaryStore, so I think it's probably better to stay in